lib/ecl/ecl_nnc_data.c

```diff
@@ -197,36 +198,37 @@ static void ecl_nnc_data_set_values(ecl_nnc_data_type * data, const ecl_grid_typ
          if (current_kw) {            
             correct_kw_count = ecl_kw_get_size( current_kw );
             check_kw_count = true;
-            data->size = nnc_index + correct_kw_count;
          }
          else {
-            check_kw_count = false;
-            printf("Warning: failed to obtain kw from file in function %s. ", __func__);
-            printf("Grid1: %d, Grid2 %d\n", current_grid1, current_grid2);
+            all_nnc_kw_present = false;
```

Comment:
Instead of setting this `bool` variable to `false` the whole function should just return `false` immediately.